### PR TITLE
Ap 4942 update merits routing copied cases

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-4942)

- Updated routing in 3 places to bypass merits task list for copied cases and go directly to the merits cya page.
- This reflects routing for passported, means-tested and non-means tested applications.
- The TaskListHelper which is used by the opponent_details partial on the merits cya page relies on legal_aid_application.legal_framework_merits_task_list however this is currently nil for copied cases. A workaround has been introduced on this ticket to obtain it from the copied application. However, a ticket has been created AP-5023 to add legal_framework_merits_task_list to the cloner service.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- The standards in the [Git Workflow document on Confluence](https://dsdmoj.atlassian.net/wiki/spaces/ATPPB/pages/4602855954/Git+Workflow) should be followed
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/services/flow/steps/provider_capital/capital_income_assessment_results_step_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  describe "#forward" do
+    subject { described_class.forward.call(legal_aid_application) }
+
+    context "when the application has been copied from another application" do
+      it { is_expected.to eq :check_merits_answers }
+    end
+
+    context "when no vehicles remain on the application" do
```
Comment (colinbruce) [inline spec/services/flow/steps/provider_capital/capital_income_assessment_results_step_spec.rb:20]: Should this description be changed?

File: spec/requests/providers/confirm_non_means_tested_applications_controller_spec.rb [hunk 2]
```diff
@@ -114,7 +114,7 @@
 
       it "updates application statre and redirects to the merits task list", :aggregate_failures do
```
Comment (colinbruce) [inline spec/requests/providers/confirm_non_means_tested_applications_controller_spec.rb:115]: The description still says a specific page

File: spec/services/flow/steps/provider_start/confirm_non_means_tested_application_step_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  describe "#forward" do
+    subject { described_class.forward.call(legal_aid_application) }
+
+    context "when the application has been copied from another application" do
+      it { is_expected.to eq :check_merits_answers }
+    end
+
+    context "when no vehicles remain on the application" do
```
Comment (colinbruce) [inline spec/services/flow/steps/provider_start/confirm_non_means_tested_application_step_spec.rb:20]: Description does not match test

**Review by colinbruce — APPROVED** [review]
LGTM :+1:

A few of the test descriptions may need tweaking :)

File: spec/services/flow/steps/provider_capital/capital_income_assessment_results_step_spec.rb (hunk 1 above)
Comment (kmahern) [reply spec/services/flow/steps/provider_capital/capital_income_assessment_results_step_spec.rb]: 👍 thanks!

File: spec/services/flow/steps/provider_start/confirm_non_means_tested_application_step_spec.rb (hunk 3 above)
Comment (kmahern) [reply spec/services/flow/steps/provider_start/confirm_non_means_tested_application_step_spec.rb]: 👍

File: spec/requests/providers/confirm_non_means_tested_applications_controller_spec.rb (hunk 2 above)
Comment (kmahern) [reply spec/requests/providers/confirm_non_means_tested_applications_controller_spec.rb]: 👍

**Pull request merged**